### ISSUE TITLE
Minimum length of name column may not be sufficient with big/wide fonts

### ISSUE THREAD

**Issue by gwhitney** [lxqt/pcmanfm-qt#1696]
##### Expected Behavior
When a directory is first opened in Detailed List view, the Name column should have adequate width to display the names of typical files in the directory.

##### Current Behavior
The Detailed List view always comes up at first with the Name column a fixed width that is the narrowest of all of the columns, so that many of the filenames are ellipsized, making the names hard to read and navigation of the file tree difficult. See attached screenshot of a typical case of this. Note the contrast of the width of the Name column with the widths of all of the other columns, which are all sized horizontally to adequately display their contents, or in some cases like Owner and Group, with more than enough horizontal width to display their contents. The contrast is starkest with the Type column, which takes up a great deal of horizontal space, enough to show its very longest item, even though it's a fairly minor piece of information and I would be happy to have Type be ellipsized to the width that Name currently is, and Name expanded to something like the width that Type currently is.

##### Possible Solution
Haven't looked at the code, no idea why Name is apparently being treated differently than and coming out much narrower than the other columns.  I work around this at the moment by dragging the name column wider, and then things look fine, but it is becoming tiresome to do this literally every time I switch to a new directory.

##### Steps to Reproduce (for bugs)
1. Open any directory  containing some files with names longer than eight characters or so, in Detailed List view.

##### Context
I am simply trying to navigate the file tree in my preferred Detailed List view without having to constantly widen the Name column.

##### System Information
* PcManFm-qt version:
```
~> pcmanfm-qt --version
pcmanfm-qt 1.2.0
```
* Distribution & Version: OpenSUSE Tumbleweed current
* Qt Version: 5.15.7
* liblxqt Version: 
   At least on Tumbleweed, pcmanfm-qt does not depend on liblxqt, and on my system, liblxqt is not installed. It does depend on libfm-qt, which is at version 1.2.0.
![Screenshot_2022-12-12_14-11-09](https://user-images.githubusercontent.com/3825429/207133452-51c69893-2e8a-40c3-8d12-45d7b407f862.png)

**Comment by tsujan** [comment]
Not reproducible.

Check whether "Auto-resize columns" is selected in the header's right click menu.

**Comment by gwhitney** [comment]
Is it possibly relevant that my screen is very high-density? (90 micron pixels, 283 dpi) Does the Name column get adequate width for you in Auto-resize mode? What could be causing the Name column to come out so narrow for me in auto-resize mode? There is definitely a problem here...

**Comment by gwhitney** [comment]
A clue, perhaps: As you can see from my screenshot, the width that my pcmanfm-qt window comes up is insufficiently wide to show the full width of all columns. If I manually resize the entire window with my window manager, as I widen it the extra width is allocated to the Name column, even to the point that if I make the window very wide, the Name column becomes much wider than it needs to be to display even the longest file name. Though all of the widening/narrowing of the application's window as a whole, every column but the Name column remains exactly fixed in width (under Auto-resize columns). If I narrow the application window as a whole, just the Name column reduces in width, until it gets down to some very narrow minimum allowed width, at which point the rightmost columns start to disappear off the right edge of the application (and a horizontal scrollbar appears at the bottom).

Thus, it seems as though only Name is being provided with any "stretchiness" in Auto-resize columns to allow the detailed list to conform to the overall application window size. In fact, as the most important column, it should reduce in size _less_ than other columns as the window is compressed, and other columns, like Type and Modified and Created, should be allowed to get narrower to conform with the overall window size. So if in fact only Name is allowed to vary in width as the space available changes, that would directly cause this buggy behavior that makes directories difficult to read.

Please could you try narrowing a window in Auto-resize mode and see if you also observe the problematic behavior that it is the Name column that becomes very narrow to the point of ellipsizing all its contents first? And if so, please reopen this bug, so that the "stretchiness" can be more equitably distributed over all of the columns in auto-resize mode? Just distributing the stretchiness equally (like the default in an HTML flexbox) should help significantly, although my intuition is that you actually want the Name column to narrow less than the other ones... Thanks for your consideration of these points!

**Comment by tsujan** [comment]
OK, the problem is the font size. In the code, the name column will be shrunken to a minimum size of 200 px if there's no space. That's not enough with big fonts.

I'll change 200 to a minimum number of characters. The report is valid.

P.S. I didn't read your last comment because it was too long.

**Comment by gwhitney** [comment]
TL; DR: It appears that an additional problem, maybe the actually critical one, is that the only column with any stretchiness in the Auto Resize mode is the the Name column, so that it takes __all__ of the hit of narrowing if the overall window is to narrow to include the natural width of all of the columns. Please try narrowing a a pcmanfm-qt window much too narrow to hold all of the columns and see if you reproduce the bad behavior. Suggested fix: distribute the "stretchiness" more fairly among all the columns in Auto-resize mode.  Thanks! And sorry for writing too much, I was trying to be super clear since my initial report seemed to be rejected.

**Comment by tsujan** [comment]
The patch is here: https://github.com/lxqt/libfm-qt/pull/857. It makes the minimum width of name column depend on the font and its size.

The eliding is still possible but not with such a small number of characters: the text will be elided if its length is greater than 50 times the space width (= 50 characters with monospace fonts; a little less with others).

**Comment by tsujan** [comment]
The PR is merged. If you don't want to compile from the source, the fix will be in 1.2.1.

Thanks for the report! And sorry that I didn't see the issue at first.
